### ISSUE TITLE
Pull request notification is triggered before changes are in destination Bitbucket repository under the refs/pull-request

### ISSUE THREAD

**Issue by James-Dengel** [tomasbjerre/pull-request-notifier-for-bitbucket#155]
So this is a strange issue and it's timing related, I think.

This is only valid for pull-requests which are from a forked repository.

I have bitbucket 4.8.3 installed with the 2.37 pull-request-notifier-for-bitbucket.
I have the following configured:
a commit notify to a jenkins server for a GIT repo which is triggered upon the pull request, operations opening, updating, and rescoped (both events)
This jenkins instance does a merge of the code from the SHA1 of the source branch to the destination.
This is configured to build from the destination repository using the following refspec
+refs/pull-requests/*:refs/remotes/origin/*
as this is how bitbucket stores the pull requests in the target repository.

Now the issue I'm seeing is that the build is triggered when a commit is made to the source repository
however the jenkins build fails because the pull request ref in the target has not been updated with the changes to the source repo, yet the pull request notification has been triggered.

This is quite advanced behaviour that I'd trying to implement, but it should work.

**Comment by tomasbjerre** [comment]
I usually have a shell script build step in Jenkins where I perform the merge myself. By fetching source and target repos and then merging source commit into target commit.

**Comment by James-Dengel** [comment]
Hey Tomas, thanks for the great work with the plugin it works great by the way.

I was just posting here to see if other had encountered this issue, it just seems strange that when opening the pull request the changes are in the 
+refs/pull-requests/

I have now found through testing that the changes are not applied to the destination utill the pull-request page in bitbucket is viewed it's only then that the changes are most in the 
+refs/pull-requests/

I guess the server only commits the changes when this is page is created/viewed.

I have raised an issue with Atlassian Support for this but I doubt that i will be able to get anywhere.
Thanks for the advice.
